app/controllers/concerns/course/lesson_plan/strategies/fomo_personalization_strategy.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+class FomoPersonalizationStrategy < BasePersonalizationStrategy
+  MIN_LEARNING_RATE = 0.67
+  MAX_LEARNING_RATE = 1.0
+  HARD_MIN_LEARNING_RATE = 0.5
+  DATE_ROUNDING_THRESHOLD = 0.8
+
+  # Shifts start_at of relevant lesson plan items and resets the bonus_end_at and end_at
+  # of the same items. The amount shifted is based the learning rate precomputed.
+  #
+  # The expected precomputed_data is the default data from precompute_data.
+  #
+  # @param [CourseUser] course_user The user to adjust the personalized timeline for.
+  # @param [Hash] precomputed_data The default data precomputed by precompute_data.
```

Comment:
I think the hash content can be indicated here (and same for the rest of the classes).

Reply:
I think no need. The hash has 3 keys with values of very different types - doesn't make sense to force it into the param type here. It'd be better to instead point to the `precompute_data` method, which has the required documentation.

Potentially, we can write out the type above, but I think there's no real need to replicate the same content from `precompute_data` to over here.